OpenMC_CSGModelMSFR/MSFR_3D_keff.py

```diff
@@ -0,0 +1,156 @@
+fuel.add_nuclide('Np237', 0.0642, 'wo')
+fuel.add_nuclide('Pu238', 0.0318, 'wo')
+fuel.add_nuclide('Pu239', 0.4393, 'wo')
+fuel.add_nuclide('Pu240', 0.2127, 'wo')
+fuel.add_nuclide('Pu241', 0.1352, 'wo')
+fuel.add_nuclide('Pu242', 0.0788, 'wo')
+fuel.add_nuclide('Am241', 0.0055, 'wo')
+fuel.add_nuclide('Am243', 0.0233, 'wo')
+fuel.add_nuclide('Cm244', 0.0092, 'wo')
+
+li_salt = openmc.Material(2, "li_salt", temperature=873.15) # these temps don't matter
+li_salt.add_nuclide('Li7', 0.9241)
+li_salt.add_nuclide('Li6', 0.0759)
+
+be_salt = openmc.Material(14, "be_salt", temperature=873.15)
+be_salt.add_element('Be', 1)
+
+f_salt = openmc.Material(114, "f_salt", temperature=873.15)
+f_salt.add_element('F', 1)
+# Ratio to get keff=1
+Ratio=0.2
+# Ratio=0.07
+# MOSART paper lists ~73 LiF 27 BeF2
+MA = 1
+Li = (MA/Ratio) * .73
+Be = (MA/Ratio) * .27
+F = MA*3 + Li + Be # This comes from chemical formula
+tot = Li + Be + MA + F
+mixList = [MA/tot, Li/tot, Be/tot, F/tot]
+mixList[3] = mixList[3] + (1-sum(mixList)) 
+fuel_salt = openmc.Material.mix_materials([fuel, li_salt, be_salt, f_salt], mixList, 'ao') # default is 'ao'
+# Define fuel salt temperature for density calculation
+FStemp = 900 # Kelvin
+# Define fuel salt density from (11) in https://www.oecd-nea.org/pt/docs/iem/jeju02/session3/SessionIII-12.pdf
+rho_salt = 3.970-(7.20*10**-4)*FStemp
+print(rho_salt)
+fuel_salt.set_density('g/cm3', rho_salt)
+# Set overall fuel salt temperature
+fuel_salt.temperature = FStemp # Original Value = 900 K
+
+# Add materials 
+mats = openmc.Materials()
+mats += [fuel, li_salt, be_salt, f_salt, fuel_salt]
+mats.export_to_xml()
+
+## Cells ## 
+
+full_MSFR_3D = openmc.Cell(name='3D_MSFR')
+full_MSFR_3D.region = full_region
+# full_MSFR_3D.region = slant_top_region | slant_bottom_region
+# full_MSFR_3D.region = core_region
+full_MSFR_3D.fill = fuel_salt
+
+## Geometry Model Setup ##
+
+root_universe = openmc.Universe(cells=[full_MSFR_3D])
+geometry = openmc.Geometry()
+geometry.root_universe = root_universe
+geometry.export_to_xml()
+
+## Plots ##
+plot = openmc.Plot()
+plot.basis = 'yz'
+plot.origin = (0,0,0)
+plot.width = (5*100,5*100)
+plot.pixels = (400,400)
+plots = openmc.Plots([plot])
+plots.export_to_xml()
+
+## Run Settings ##
+
+# Set the source/k-code settings
+settings=openmc.Settings()
+# Set the source type and location
+point = openmc.stats.Point((0,0,0))
+source = openmc.Source(space=point)
+settings.source = source
+# settings.batches = 250
+settings.batches = 400
+settings.inactive = 100
+settings.particles = 10000
+# settings.particles = 5000
+## Use previous yaml (from Josh)
+# settings.run_mode = 'fixed source'
+settings.temperature = {'method' : 'interpolation'}
+settings.export_to_xml()
+
+tally2 = openmc.Tally(2)
+# tally2.filters = [cell_filter]
+tally2.scores = ['heating', 'fission']
+# Export tallies
+# tallies = openmc.Tallies([tally1, tally2])
+tallies = openmc.Tallies([tally2])
+tallies.export_to_xml()
+
+openmc.run()
```

Comment:
I think it would be good to add a few functions to split this up. I'm envisioning a function that returns an `openmc.model` object and a separate function (maybe `main()`) that calls that function, runs `model.export_to_xml()` and then does `openmc.run()`. That function could also parse any command line arugments. We could also have the model generating function call other functions